\chapter*{Introduction}
\addcontentsline{toc}{chapter}{Introduction}

Gene regulatory networks tend to be crucial in various genomic studies such as gene expression, cell segregation and differentiation, and disease~\cite{collas2010current}. 
Transcription factors (TFs) and chromatin modifiers are the main elements that control cellular functions by the dynamic association with target DNA within regulatory regions such as promoters and enhancers and their coding sequences.
Epigenetic modifications change the composition of the chromatin and modulate the association of the regulatory elements with DNA~\cite{antequera2003structure, kouzarides2007chromatin, mito2007histone}.
Several methods have been developed to identify biological regulatory elements and their behavior in the context of the interaction with chromatin and other transcription regulation components. 

Chromatin immunoprecipitation, known as ChIP, is a technique to investigate and analyze any protein associated with DNA inside the cell~\cite{o1995histone, o1996immunoprecipitation, nelson2006protocol}.
The most frequently investigated chromatin-associated proteins are transcription factors, post-translationally modified histones and histone variants in the genome, components of the transcriptional machinery, and chromatin-modifying enzymes.
Classical ChIP assays require a large number of cells because of the loss of material and extensive sample handling which leads to errors and inconsistency between replicates~\cite{o1996immunoprecipitation}. 
However, the introduction of the newer ChIP protocols promotes simplification of the procedure. 
Those simplifications enable the technique to be applied for a small amount of the cells and discover the genome-wide profiling of the target elements without prior knowledge of exact binding loci~\cite{collas2010current}. 

DNA microarrays in combination with ChIP assay was the first approach to identify and analyze \textit{in-vivo} the protein-chromatin interactions of interest in broad genome coverage. 
Such an approach is known as chromatin immunoprecipitation followed by genomic tiling microarray hybridization or simply ChIP-on-chip~\cite{ren2000genome, loden2005whole}. 
This technology produces highly reproducible profiles.
However, the large number of false positives and several limitations, such as genomic coverage, which is dependent on the microarray probe design, stimulated the development of other strategies. 

Chromatin immunoprecipitation followed by sequencing, known as ChIP-sequencing or ChIP-seq, is a modern and relatively cheap method to analyze any protein associated with DNA. 
Advances in high-throughput parallel sequencing technology and computational methods enable to generate and analyze extremely large data sets. 
Unlike Chip-on-chip, ChIP-seq generally builds a profile with better resolution and signal-to-noise ratio, and detect more signals~\cite{park2009chip}.
Also, the advantage of the ChIP-seq protocol is the small amount of the initial material~\cite{adli2010genome}.