\chapter{Chromatin immunoprecepitation}
\section{Experiment design}
Chromatin immunoprecepitation followed by sequencing (ChIP-seq) is a method applied on a genome wide scale identification of transcription factor (TF) binding sites and epigenetic marks.
Genome-wide studies using ChIP-seq provide important insights into biological role of presence of the histone modification and/or transcription factors.
ChIP assay involves immunoprecepitation of proteins of interest along with assosiated DNA.
The proteins are crosslinked to DNA using  \emph{in vivo} formaldehyde, and then the chromatin is fragmented either by micrococcal nuclease or sonication.
A typical experiment requires 10-100 ng od chromatin.
Before the purification point, the target proeins incubated with specific antibody.
After washing, the uncrosslinced and purified DNA can be taken to the library construction.

\section{MOCK versus input}


\section{Sequencing library quality plays the critical role for next-generation sequencing}
The fragmentation of the target DNA is a key step for NGS library construction.
Physical metods or enzymatic methods are typically used for DNA fragmentation~\cite{}.
%comparison of both phys and enzyme?
After fragmentation the sequencing oligonucleotyde adapters of a constant length with specific sequences are ligated.
Those sequences are specially designed to interact with the NGS platform such as Illumina or Ion Torrent.
Determination of the library size referring to the insert size which is limited by the high-throughput sequencing instrumentation.
Process of cluster generation, after the fragments have attached, constrains the optimal insert size in Illumina technology.
Shorter products amplify more efficiently.

Library preparation from purified DNA for ChIP-seq experiment should be as complex as possible.
More starting material induces less amplification.
Thus library complaxity is better.



\section{Sequencing}
Purified DNAs fragments from ChIP-seq samples are sequenced as reads of length 36-100bp and can be uniquely aligned to the genome reference.
Different next-generation platforms can be utilized.
The technology provides siquencing of both single-end and paired-end reads.
A single run of the Illumina/Solexa sequencing technology generetas 50-200 million reads\cite{park2009chip}.
Single-end reads are usrd for common ChIP-seq analysis, while paired-end are evolved to improve the library complexity.
That helps to focuse at repetitive regions \cite{chen2012systematic}.

\section{Mapping}
Sequenced short reads are represented as a text file and can be aligned with mapping software based on Burrows-Wheeler transform algorithm \cite{li2009fast} \cite{siren2014indexing}
