\chapter{Sources of known biases and artefacts in ChIP-seq experiments.}



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{Effect of crosslinking time.}
\label{formaldehyde}

%\todo{..., effect/impact of crosslinking time?}
%ChIP-seq technology has been used to identify the localization of post-translationally modified histones, histone variants, TF, and other chromatin-associated proteins.

Native ChIP is commonly used for the analysis of stable chromatin complexes~\cite{kasinathan2014high} and histones.
Transcription factor interactions that would not withstand the isolation procedure due to weak affinity to the DNA require a crosslinking step.
%\todo{tento odstavec se dubluje s obsahem kapitoly 1.2}
% \todo{"native" bych zde nepoužíval, spíš specific nebo genuine. Nativní se používá jako protiklad k fixovaný (např. native vs formaldehyde-fixed ChIP)}

Typical ChIP assay involves crosslinking using formaldehyde reactivity. 
This smallest aldehyde has been applied for decades in numerous fields~\cite{eckels2003formalin,werner2000effect,gavrilov2015vivo}.
In the chromatin immunoprecipitation, the ability of formaldehyde to crosslink with proteins and DNA to form protein-DNA or protein-protein complexes provides identification of the location of transcription factor binding along the DNA strand.
Cell permeability, short spacer length, rapid reactivity, and reversibility are key properties that define formaldehyde as a crosslinking fixative of choice for ChIP~\cite{hoffman2015formaldehyde}.
%\todo{tahle věta je HODNĚ podobná původnímu textu v článku. Dávej na podobné věci VELKÝ pozor. Měla bys vše popisovat vlastními slovy. Nejde doslovně přebírat celé věty (a prohození 2 slov to nezachrání), i když cituješ zdroj. Mohlo by to být hodnoceno jako plagiátorství. Obzvlášť, když s angličtinou trochu bojuješ a najednou se v tvém textu objeví věta nebo dokonce celý odstavec s perfektní angličtinou, vypadá to podezřele. Z ničeho tě neobviňuju, jen si to prosím hlídej.}

Unlike other essential regulatory proteins, transcription factors contain at least one DNA-binding domain that contains at least one structural motif that specifically recognizes dsDNA or ssDNA~\cite{mitchell1989transcriptional}.
Some transcription factors bind non-specifically~\cite{struhl2007interpreting};
and it is also known, that site-specific TFs can bind non-specifically~\cite{hammar2012lac,mirny2009protein} but with weaker affinity.

Several studies show that different DNA-binding proteins spend some time non-specifically bound and sliding along DNA~\cite{slutsky2004kinetics,mirny2010nucleosome,cherstvy2008protein,hu2006proteins,sheinman2009effects}.
And the approach is that the search process is a combination of three-dimensional and one-dimensional diffusion and also can be statistically predicted~\cite{sela2011dna}.
Such non-specific accessibility of the genome in different regions leads to the non-specific cross-linking events, which can increase the rate of false-positive signals in further analysis.

The formaldehyde concentration, incubation time, and other conditions can vary from experiment to experiment. 
Even though the effective crosslinking time is of the 5-second threshold~\cite{schmiedeberg2009temporal}, many laboratories use rather prolonged formaldehyde incubation time.
\todo{buď konkrétní - jaké časy se používají?}
Soluble proteins not expected to associate with DNA would be non-specifically captures at the open chromatin loci. Shortening the incubation time traps preferentially genuine DNA-protein binding events~\cite{baranello2016chip}.

Antibody-beads also can non-specifically bind cross-linked complexes during the immunoprecipitation step~\cite{zhu2014fast} and cause a raise of false-positive rate.
It is important to mention, such bias can appear due to formaldehyde excess~\cite{hanson2018using}.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%5
\section{Hyper ChIPable regions are another source of false-positive signals.}

Another challenge the researcher can face during the ChIP-seq analysis is that some regions are more susceptible to immunoprecipitation. 
Those areas may show a strong signal for the loci where a protein of interest has no binding event~\cite{teytelman2013highly}.
As was described in the previous section~\ref{formaldehyde}, crosslinking using formaldehyde may trap mobile proteins in areas with promoters of actively transcribed genes. 
Such promoters are localized in regions of open chromatin where high transcription factor occupancy.  
That can explain the presence of a ChIP signal. 
However, if there is no antigen in the sample, but the signals are still present in the ChIP profile, the formaldehyde crosslinking bias cannot explain such a fact~\cite{jain2015active}.

The term high-occupancy target (HOT) region is used to describe sites of highly expressed genes~\cite{teytelman2013highly}.  
Such loci are characterized by the occupancy of multiple unrelated TFs and transcription machinery proteins~\cite{boyle2014comparative}. 
Unstructured globule surface of proteins sometimes caused by low sequence complexity may interact non-specifically with beads or antibodies during immunoprecipitation~\cite{teytelman2013highly}.
TFs in protein complexes reach specific conformation only during the interaction between components. 
That specificity can be disrupted (for example, by sonication) and exhibit unstructured surface, which after IP leads to a hyper-ChIPability~\cite{jain2015active}.
Several studies have described that phenomenon in different organisms~\cite{boyle2014comparative, fan2009does, krebs2014optimization, teytelman2013highly,   waldminghaus2010chip, xie2013dynamic}.

The HOT region location is typically at promoter locus of highly expressed genes close to the transcription start site (TSS), and that genes stand behind essential cellular functions~\cite{wreczycka2019hot}. 
One possible solution is to use the so-called Knockout Implemented Normalization (KOIN) method, where the protein is not present~\cite{krebs2014optimization}. 
This method significantly improves false-positive identification, which can help more accurate downstream interpretation during analysis~\cite{wreczycka2019hot}.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\section{GC content bias in ChIP-seq is challenging.}

Effective analysis requires sufficient coverage by sequence reads. 
The sequenced reads must overlap to achieve the required level of resolution for a ChIP experiment.
%\todo{tuhle větu nechápu - proč se ready musí překrývat?}
Whereby the experiment requires many copies of a whole genome.
Nowadays ChIP purified DNA is obtained from many cells.
Multiple genome copies provide a sufficient number of overlapping reads after cleavage and sequencing.
%\todo{"after fragmentation", spíš než "after cleavage"}
In addition, sigle-cell sequencing requires PCR-amplification before the fragmentation step~\cite{clark2016single}.
%\todo{vysvětli stručně, proč je zde třeba materiál amplifikovat}

% \todo{důležitá poznámka k celému odstavci: DNA/chromatin se u ChIPu fragmentuje pouze před imunoprecipitací. Poté, co vyizoluješ čistou DNA, se už žádná další fragmentace neprovádí. Rovnou se připraví knihovna a sekvenuje se!}

The coverage for ChIP-seq varies across experiments due to GC-content bias, and that variability may introduce false-positive signals during downstream analysis.
%\todo{podle mě je důležitější, že "The coverage for ChIP-seq varies across the genome due to GC-content bias..."}
Loci with high GC base composition are often under-represented, and such bias has been observed in several NGS experiments~\cite{benjamini2012summarizing,dohm2008substantial,teng2017accounting}.
This bias tends to have appeared during the PCR amplification~\cite{ross2013characterizing}.
It is known that GC-rich DNA targets are less amenable to amplification due to base base-stacking interations~\cite{yakovchuk2006base}.
% \todo{GC bias vzniká také v průběhu PCR, kdy se GC-rich fragmeny amplifikují hůře než GC-poor. Je to dáno jejich méně efektivní denaturací - mají vyšší vazebnou energii}

Both PCR amplification during library construction and PCR for cluster amplification on the Illumina platform play role. 
%\todo{"... PCR for cluster generation..."}
The first case has many potential solutions to avoid the bias during library construction step.
It requires preparing a larger amount of the input DNA to avoid PCR itself, or PCR step with extensive optimization such as annealing time and temperatures and primer specifity~\cite{aird2011analyzing}.
The possible problem of Illumina cell flow bridge amplification may occur due to secondary structures.
%\todo{"cell flow" → flowcell}
GC-rich sequences form hairpins, which are stable and stack unmelted at usual PCR denaturation temperatures~\cite{stein2010nucleosome}.

Fragmentation is the first step in the DNA sequencing process.
This step ensures the solubility of the crosslinked DNA-protein complexes are soluble and accessible to antibody.
Using enzymatic digestion, chemical shearing, or different mechanical method, DNA with crosslinked proteins is broken up into short fragments.
After immunoprecipitation one or both ends of the fragment sequenced.
%\todo{připadá mi, že na tomto místě textu už je pozdě detailně vysvětlovat metody fragmentace. To se má čtenář dozvědět dřív. Zde můžeš třeba už jen odkázat na příslušnou předchozí kapitolu.}

The early assumption was that genomic DNA break randomly.
However, the experiment~\cite{poptsova2014non} with different methods of DNA fragmentation showed that the rates of double-stranded breaks are sequence-dependent.
This dependency may also cause the GC-bias. 

However, true binding sites of the protein of interest may be expected to occur in high GC regions.
% \todo{"are expected" -> spíš "may be expected"}\todo{. Oblasti vazby vůbec nemusí být GC-bohaté, záleží na cílovém proteinu}
Those regions have biological relevance such as promotor regions
%\todo{jen připomínám, že tohle se liší podle organismu. Třeba S. pombe má promotory GC-poor}.
Modeling of the GC-content at the fragment level~\cite{benjamini2012summarizing} is the optimal approach, but is not directly applicable to ChIP-seq analysis.
The method for correction of this kind of bias was presented~\cite{teng2017accounting}.
Incorporation of the approach into current peak callers shows substantial improvements in signal finding.
However, the method is not suitable for broad peak identification as histone modifications.


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 \section{Problematic regions of the genome.}

 Accurate annotation of the genome assembly is required to get proper information after mapping of the sequenced reads. 
 Unfortunately, some genomes may contain misassembled and gap regions due to repetitive sequences~\cite{amemiya2019encode}.
 Recent long-read sequencing technologies~\cite{feng2015nanopore} and newly developed software improve more accurate \textit{de novo} assembles~\cite{rice2019new}. 
 Even though new assemblies are generated using accurate modern methods, such under-represented regions in the reference can bias the ChIP-seq results and produce an ambiguous outcome~\cite{carroll2014impact}. 
 One possible solution is to detect and remove such regions when analyzing data to improve the assessment of biologically relevant signals during normalization and peak calling~\cite{amemiya2019encode}. 


 %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
 \section{Input versus MOCK}
 \label{control}

 Earlier in section \ref{strategy} was mentioned about two basic types of control data. 
 Cross-linked and fragmented DNA under the same conditions as IP is referred to as an input control dataset. 
 The second one is IgG/GFP control (also known as mock), where IP lacks specific interactions due to untagged with epitope TF or using antibodies, which cannot recognize the TF~\cite{landt2012chip, shin2013computational}.
 The theoretical distribution of noise is expected to be uniform~\cite{robertson2007genome}. 
 However, existing technical and biological biases require either input or mock control dataset.

 A small number of reads and PCR duplicates, which appears due to limited genomic regions, results in problematic normalization, making mock control less favorable than an input~\cite{kidder2011chip}. 
 Direct sequencing of crosslinked chromatin without IP contains the bias of sonication and is able to normalize that bias~\cite{kharchenko2008design}. 
 Therefore, input control is recommended to distinguish real signals from artifacts. 
 Also, input datasets produced from the same cell line under the same circumstances may be combined~\cite{meyer2014identifying}.

 Even if the input is preferred to mock due to library complexity, there is still the presence of biases, which are not able to get fid of using only input control. 
 Expression and open chromatin biases appear in the ChIP-seq experiment, and the mock IP as control can correct such a factor~cite{park2013widespread}.

 In addition to both input and mock normalization methods, there is also spike-in normalization, in which foreign chromatin is used as internal control and helps avoid several biases~\cite{bonhoure2014quantifying}.

