\chapter{bias, artefacts}

% To ensure reliability of the data, at least duplicate biological replicate experiments should be done.
% Sonication of crosslinked chromatin may be preffered method.
% Mocrococcal nuclease degrade linker DNA where TF tend to bind.
% The conditions of sonication should be optimized for each cell type.
% Because they depend on cell type, type of sonicator and  sonicator settings.

\section{assay artefacts}
Strong enrichment signals suggestive of proteins binding to genomic loci where genes were higly transcribed were found.
Moreover, the enrichment for proteins binding to higly-transcribed genes was was observed even in controls like moch ChIP-seq data.
Which poins to an overall bias that could contaminate any ChIP-seq data with false positives~\cite{park2013widespread}
A secondary bias of nucleosomal periodicity was also commonly observed across ChIP-seq dataset.
And contributed additional false positives in which proteins falsly appeared to interact with nucleosomes.

Two features among strong false positive signals.
First, the signals were present within gene bodies.
Second, Stronges signal derived from genes thet are known to be higly expressed.

% It is possible that certain TF trully bind to ORFas a means of regulating gene expression.
A common use of ChIP-seq is to examine binding of a given factor under different growth conditions or backgrounds.
Since only a single variable is changed (growth condition), it might be assumed that comparing binding under different conditions offers a reliable means of identifying biologically relevant targets, with most background artefacts being ormalized out.

One bias arised during genome sonication.
Open chromatin regions are easely shared than other regions.
Thus these open regions yield more protein-DNA complexes.
IP step immunoprecepitate more complexes from the open chromatin regions.
And as a result gives more sequenced reads.

To correct this bias, the fragmented genome are divided into two portions.
One portion goes through the IP step.
And then sequenced.
Other portion is sequenced directly.
This portion serve the input control.

This input control can be used to normalize the shearing bias of sonication~\cite{kharchenko2008design}.
% Type of normalization controls might be appropriate for normalizing false positives?

In the analysis of ChIP-seq data two types of normalization or correction controls are commonly used.

The input sample has the advanage that all the regions of the genome are well represented.
The sample concentration is ample and stable for constructing sequencing libraries.
The same sample can potentially serve as the control for several related experiments.

The input a baseline signal for reads across the genome, factoring in sequence mappability and copy number differences relative to the reference genome.
For these reasons, input has been suggested as a more effective control.

However, genes transcribed at high rates is not adequately represented in the input.

A mock ChIP is designed to correct data with large quantity of spurious sites in ChIP-seq, which are coused by uneven genomic sonication and nonspecific interactions between chromatin and antibody.
\todo{nezapomeň na nespecifické interakce mezi chromatinem a magnetickými/polysacharidovými kuličkami, které se používají pro purifikaci komplexů chromatin-protilátka}
Whereas DNA input controls corrects only for uneven sonication.
\todo{řekl bych, že input dokáže korigovat i PCR amplification bias (GC bias?)}

Mock better reflects the background enrichment from highly transcribed genes.
Mock ChIP-seq data exhibit a stronger expression bias than the corresponding input sample.
Therefore correction by mock ChIP (normalization) would be more effectively reduce the false-positives than normalization by input~\cite{park2013widespread}.
Mock ia s better control for minimizing the appearence of occupancy signal over transcribed regions.
\todo{mock IP má ale tendenci precipitovat jen velmi malé množství chromatinu (DNA), které je do určité míry náhodné a tedy nereproducibilní. Při PCR pak dochází k overamplifikaci.}

Measuring the binding of a TF under two different conditions and identifying the differentially bound target offers the most reliable way to identify targets of biological significence.
\todo{s tímhle nesouhlasím, je to příliš obecně definované. Pokud například opůsobím buňky nějakým inhibitorem, který do 10 min způsobí disociaci TF z chromatinu, tak se to na transkripci moc neprojeví a artefakty mi zůstanou prakticky stejné.}
The assumption that most sources of background signal are canseled out between two samples is risky.
Expression biase derives directly from actively transcribed genes.
Transcription will differ between two conditions.
\todo{není jisté, zda se bude transkripce mezi podmínkami lišit. Záleží, jaké to jsou podmínky. A určitě bude existovat nějaká skupina vysoce exprimovaných housekeeping genů, které budou silně transkribované za obou podmínek: např. proteiny translačního aparátu, cytoskelet atd.}
Even in these cases ChIP data from each condition has to be properly corrected by the corresponding mock ChIP data to minimize false positives.

% CROSSLINK
% ===========
\section{Non-specific TF interactions, crosslinking time}

ChIP-seq technology has been used to identify the localization of post-translationally modified histones, histone variants, TF, and other chromatin-associated proteins.
Native ChIP is commonly used for the analysis of stable chromatin complexes~\cite{kasinathan2014high} and histones.
Transcription factor interactions that would not withstand the isolation procedure due to weak affinity to the DNA require a crosslinking step.
% \todo{"native" bych zde nepoužíval, spíš specific nebo genuine. Nativní se používá jako protiklad k fixovaný (např. native vs formaldehyde-fixed ChIP)}


Typical ChIP assay involves crosslinking using formaldehyde reactivity. 
This smallest aldehyde has been applied for decades in numerous fields~\cite{eckels2003formalin,werner2000effect,gavrilov2015vivo}.
In the chromatin immunoprecipitation, the ability of formaldehyde to crosslink with proteins and DNA to form protein-DNA or protein-protein complexes provides identification of the location of transcription factor binding along the DNA strand.
Cell permeability, short spacer length, rapid reactivity, and reversibility are key properties that define formaldehyde as a crosslinking fixative of choice for ChIP~\cite{hoffman2015formaldehyde}.

Unlike other essential regulatory proteins, transcription factors contain at least one DNA-binding domain that contains at least one structural motif that specifically recognizes dsDNA or ssDNA~\cite{mitchell1989transcriptional}.
Some transcription factors bind non-specifically~\cite{struhl2007interpreting};
and it is also known, that site-specific TFs can bind non-specifically~\cite{hammar2012lac,mirny2009protein} but with weaker affinity.

Several studies show that different DNA-binding proteins spend some time non-specifically bound and sliding along DNA~\cite{slutsky2004kinetics,mirny2010nucleosome,cherstvy2008protein,hu2006proteins,sheinman2009effects}.
And the approach is that the search process is a combination of three-dimensional and one-dimensional diffusion and also can be statistically predicted~\cite{sela2011dna}.
Such non-specific accessibility of the genome in different regions leads to the non-specific cross-linking events, which can increase the rate of false-positive signals in further analysis.



% Even low-level non-specific interactions between the antibody and cross-reacting cellular proteins may contribute to this phenomenon.
% Or Open chromatin shows preferential recovery through the immunoprecipitation process.
% Highly expressed genes are characterized by open chromatin conformation, nucleosome perturbation and DNA more exposed to interactions.\cite{}
% Although the antibody in IP binds specifically to its target TFs.
% It can also bind non-specifically to other proteins.
% Under some cross-linkin conditions, this regions might cause the hyper ChIPability\cite{}

% %\todo{jak už jsem psal výše, hlavím zdrojem nespecifických interakcí je adsorpce na povrch kuliček. Protilátka může a nemusí také přispívat}

% One possible solution is to develop alternative approaches that avoid the cross-linking step.
% The lack of cross-linking necessarily means that only proteins very tightly associated with the chromatin can be immunoprecipitated.
% Non-specific cross-linking in the human cell line is not an issue if formaldehyde treatment is limited to 10 min or less~\cite{}.
% Shortening the fixation time reduces the background level due to non-specific protein binding to trap preferentially native DNA-proteins interactions~\cite{baranello2016chip}.

% Under some conditions, cross-linked artifacts might be relevant.
% Good biochemical practice and proper controls can detect and minimize this bias.

% To control for nonspecific antibody bias, a mock ChIP can be utilized.

% However, the mock ChIP yields much less DNA material than DNA input.
% And mock is more susceptible to technical noise~\cite{kidder2011chip, landt2012chip}.

% GC bias
% ============
\section{GC content bias in ChIP-seq is challenging}
Effective analysis requires sufficient coverage by sequence reads. 
The sequenced reads must overlap to achieve the required level of resolution for a ChIP experiment.
Whereby the experiment requires many copies of a whole genome.
Nowadays ChIP purified DNA is obtained from many cells.
Multiple genome copies provide a sufficient number of overlapping reads after cleavage and sequencing.
In addition, sigle-cell sequencing requires PCR-amplification before the fragmentation step~\cite{clark2016single}.

% \todo{důležitá poznámka k celému odstavci: DNA/chromatin se u ChIPu fragmentuje pouze před imunoprecipitací. Poté, co vyizoluješ čistou DNA, se už žádná další fragmentace neprovádí. Rovnou se připraví knihovna a sekvenuje se!}

The coverage for ChIP-seq varies across experiments due to GC-content bias, and that variability may introduce false-positive signals during downstream analysis.
Loci with high GC base composition are often under-represented, and such bias has been observed in several NGS experiments~\cite{benjamini2012summarizing,dohm2008substantial,teng2017accounting}.
This bias tends to have appeared during the PCR amplification~\cite{ross2013characterizing}.
It is known that GC-rich DNA targets are less amenable to amplification due to base base-stacking interations~\cite{yakovchuk2006base}.
% \todo{GC bias vzniká také v průběhu PCR, kdy se GC-rich fragmeny amplifikují hůře než GC-poor. Je to dáno jejich méně efektivní denaturací - mají vyšší vazebnou energii}

Both PCR amplification during library construction and PCR for cluster amplification on the Illumina platform play role. 
The first case has many potential solutions to avoid the bias during library construction step.
It requires preparing a larger amount of the input DNA to avoid PCR itself, or PCR step with extensive optimization such as annealing time and temperatures and primer specifity~\cite{aird2011analyzing}.
The possible problem of Illumina cell flow bridge amplification may occur due to secondary structures.
GC-rich sequences form hairpins, which are stable and stack unmelted at usual PCR denaturation temperatures~\cite{stein2010nucleosome}.

Fragmentation is the first step in DNA sequencing process.
This step ensures the solubility of the crosslinked DNA-protein complexes are soluble and accessible to antibody.
Using enzymatic digestion, chemical shearing, or different mechanical method, DNA with crosslinked proteins is broken up into short fragments.
After immunoprecipitation one or both ends of the fragment sequenced.


The early assumption was that genomic DNA break randomly.
However, the experiment~\cite{poptsova2014non} with different methods of DNA fragmentation showed that the rates of double-stranded breaks are sequence-dependent.
This dependency may also cause the GC-bias. 

However, true binding sites of the protein of interest may be expected to occur in high GC regions.
% \todo{"are expected" -> spíš "may be expected"}\todo{. Oblasti vazby vůbec nemusí být GC-bohaté, záleží na cílovém proteinu}
Those regions have biological relevance such as promotor regions.
Modeling of the GC-content at the fragment level~\cite{benjamini2012summarizing} is the optimal approach, but is not directly applicable to ChIP-seq analysis.
The method for correction of this kind of bias was presented~\cite{teng2017accounting}.
Incorporation of the approach into current peak callers shows substantial improvements in signal finding.
However, the method is not suitable for broad peak identification as histone modifications.






% Blacklist
% ============
\section{Blacklist region/Difficulties in the ussambly/overrepresented regions}

Signals based on high-throughout sequencing output rely on an accurate genomic annotation and mapping.
In some regions, such as regions with a large number of structural variants and repetitive regions, the assembly of the genome can be difficult and may may be collapsed or under-represented relative to the actual underlying genomic sequence.
Problems with the assembly have effects on the mapping and calling and lead to inaccurate interpretation of the downstream analysis.
Duplication rate for single-end ChIP-seq data is overestimated and leads to ChIP-seq signal being lost within those regions~\cite{chen2012systematic}.
