\chapter{bias, artefacts}

% To ensure reliability of the data, at least duplicate biological replicate experiments should be done.
% Sonication of crosslinked chromatin may be preffered method.
% Mocrococcal nuclease degrade linker DNA where TF tend to bind.
% The conditions of sonication should be optimized for each cell type.
% Because they depend on cell type, type of sonicator and  sonicator settings.

\section{assay artefacts}
Strong enrichment signals suggestive of proteins binding to genomic loci where genes were higly transcribed were found.
Moreover, the enrichment for proteins binding to higly-transcribed genes was was observed even in controls like moch ChIP-seq data.
Which poins to an overall bias that could contaminate any ChIP-seq data with false positives~\cite{park2013widespread}
A secondary bias of nucleosomal periodicity was also commonly observed across ChIP-seq dataset.
And contributed additional false positives in which proteins falsly appeared to interact with nucleosomes.

Two features among strong false positive signals.
First, the signals were present within gene bodies.
Second, Stronges signal derived from genes thet are known to be higly expressed.

% It is possible that certain TF trully bind to ORFas a means of regulating gene expression.
A common use of ChIP-seq is to examine binding of a given factor under different growth conditions or backgrounds.
Since only a single variable is changed (growth condition), it might be assumed that comparing binding under different conditions offers a reliable means of identifying biologically relevant targets, with most background artefacts being ormalized out.

One bias arised during genome sonication.
Open chromatin regions are easely shared than other regions.
Thus these open regions yield more protein-DNA complexes.
IP step immunoprecepitate more complexes from the open chromatin regions.
And as a result gives more sequenced reads.

To correct this bias, the fragmented genome are divided into two portions.
One portion goes through the IP step.
And then sequenced.
Other portion is sequenced directly.
This portion serve the input control.

This input control can be used to normalize the shearing bias of sonication~\cite{kharchenko2008design}.
% Type of normalization controls might be appropriate for normalizing false positives?

In the analysis of ChIP-seq data two types of normalization or correction controls are commonly used.

The input sample has the advanage that all the regions of the genome are well represented.
The sample concentration is ample and stable for constructing sequencing libraries.
The same sample can potentially serve as the control for several related experiments.

The input a baseline signal for reads across the genome, factoring in sequence mappability and copy number differences relative to the reference genome.
For these reasons, input has been suggested as a more effective control.

However, genes transcribed at high rates is not adequately represented in the input.

A mock ChIP is designed to correct data with large quantity of spurious sites in ChIP-seq, which are coused by uneven genomic sonication and nonspecific interactions between chromatin and antibody.
Whereas DNA input controls corrects only for uneven sonication.

Mock better reflects the background enrichment from highly transcribed genes.
Mock ChIP-seq data exhibit a stronger expression bias than the corresponding input sample.
Therefore correction by mock ChIP (normalization) would be more effectively reduce the false-positives than normalization by input~\cite{park2013widespread}.
Mock ia s better control for minimizing the appearence of occupancy signal over transcribed regions.

Measuring the binding of a TF under two different conditions and identifying the differentially bound target offers the most reliable way to identify targets of biological significence.
The assumption that most sources of background signal are canseled out between two samples is risky.
Expression biase derives directly from actively transcribed genes.
Transcription will differ between two conditions.
Even in these cases ChIP data from each condition has to be properly corrected by the corresponding mock ChIP data to minimize false positives.

\section{crosslink bias}

Source and mechanism of the background expression bias arise from direct or indirect non-specific interactons of the immunoprecepitated protein with DNA in  open chromatin at highly transcribed regions, trapped by the crosslinked process.
It is unclear why the phenomenon exists in mock ChIP datasets.
It is possible that even low level non-spesific interactions between the antibody and cross-reacting cellular proteins contrebute to this phenomenon.
Or Open chromatin shows oreferential recovery through the immunoprecepitation process.
Highly expressed genes are characterized by open chromatin conformation, nucleosome perturbation and DNA more exposed to interactions.\cite{}
Although the antibody in IP binds specifically to its target TFs.
It can also bind non-specifically to other proteins.
Under some cross-linkin conditions, this regions might cause the hyper ChIPability\cite{}

One possible solution is to develope alternative approaches that avoid the cross-linking step.
The lack of cross-linking necessarily means that only proteins very tightly assosiated with the chromatin can be immunoprecepitated.
Non-specific cross-linking in human cell line is not an issue if formaldehyde treatment is limited to 10 min or less~\cite{}.
Shortering the fixation time reduces the background level due to non-specific protein binding to trap preferentially native DNA-proteins interactions~\cite{baranello2016chip}.
Under some conditions cross-linked artefacts might be relevant.
Good biochemical practice and proper controls can detect and minimize this bias.

To control for nonspecific antibody bias, a mock ChIP can be utilized.

However, mock ChIP yields much less DNA material than DNA input.
And mock is more susceptible to technical noise~\cite{kidder2011chip, landt2012chip}.

%%%
\section{GC content bias in ChIP-seq is challenging}

Fragmentation is the first step of DNA sequencing process.
Using enzymatic digestion, chemical shearing or different mechanical method, purified DNA is broken up into short fragments.
The sequenced reads must overlap to achieve the required level of resolution for a ChIP experiment the genome regions should be sequenced multiple times.
Whereby the experiment requires many copies of a whole genome.
Nowadays ChIP purified DNA is obtained from many cells.
Multiple genome copies provide a sufficient number of overlapping reads after cleavage and sequencing.
In addition, sigle-cell sequencing requires PCR-amplification before the fragmentation step~\cite{clark2016single}.

The early assumption was that genomic DNA break randomly.
However, the experiment~\cite{poptsova2014non} with different methods of DNA fragmentation showed that the rates of double-stranded breaks are sequence-dependent.
Which can lead the source of the GC-bias have been observed in several NGS experiments~\cite{benjamini2012summarizing}~\cite{dohm2008substantial}. 

The coverage for ChIP-seq varies across experiments due to GC-content bias.
And that variability leads obtain false-positive signals.
However, true binding sites of the protein of interest are expected to occour in high GC regions.
Those regions have biological relevance such as promotore regions.
Modelling of the GC-content at the fragment level~\cite{benjamini2012summarizing} is the optimal approach, but is not directly applicable to ChIP-seq analysis.

The mixed-model method for the GC-content bias correction was presented~\cite{teng2017accounting}.
Incorporation of the approach into current peak callers shows substantial improvements in signal finding.
However, the method is not suitable for broad peak identification as histone modifications.

Also new methods of cleavege of the purified DNA cleavage for library construction are developing~\cite{}, which provide better uniform coverage for GC reagions.


% It is troubling that the number of reads mapped to a genome region depends on the sequence itself.

% Windows with a GC content of 40\% contain almost twice as many reads as windows with lower content~\cite{dohm2008substantial}.
% This variability does not reflect the signal of interest, but might confuced it.
% Since GC abundance is heterogeneous across the genome and often correlated with functionality, the GC effect can be hard to tell apart from the true signal.
% The effect is not consistent between repeated experiments, or even libraries within the same experiment.
% GC counts could be associated with the stability of the DNA, and thus modify the probability of a fragmentation point occurring in the genome.
% Binding sites are expected to occur in or near high GC-content regions such as gene promoters.
% % IDR algorithms to select peaks
% GC-content bias lead peak callers to report a substantial number of false positives.

% Most peak callers operate on bin level information.
% Algorithms define bins, calculate coverage measurements.
% % something about new approach using a mixed model, which permits independent adjustments of the signal and back- ground signals and thus circumvents the confounding challenge and can be incorporated into most current peak callers


